### PR TITLE
clear creep path upon each action assignment

### PR DESCRIPTION
**Pull request by cyberblast**
solving #529

### CODE REVIEW

File: population.js [hunk 1]
```diff
@@ -87,6 +91,11 @@ mod.registerAction = function(creep, action, target, entry) {
             target.targetOf = [entry];
         else target.targetOf.push(entry);
     }
+    // clear saved path
+    if( targetId != oldTargetId ) {
```
Comment (karlthepagan) [inline population.js]: also compare actionId here?

Comment (cyberblast) [inline population.js]: why?

Comment (karlthepagan) [inline population.js]: I'm not familiar enough to know if multiple actions could have the same target. I'll take a closer look.

For example, if a privateer going to uncharge storage is manually assigned to dismantle.

Comment (cyberblast) [inline population.js]: callign assign will always assign a target as well (defined as argument or it will call newTarget). Same target same path... ;)

Comment (cyberblast) [inline population.js]: callign assign will always assign a target as well (defined as argument or it will call newTarget). Same target same path... ;)

**Pull request merged**